includes/Core/Assets/Assets.php

```diff
@@ -728,11 +731,33 @@ private function run_before_print_callbacks( WP_Dependencies $dependencies, arra
 
 			if ( isset( $this->assets[ $handle ] ) ) {
 				$this->assets[ $handle ]->before_print();
+
+				if ( $is_amp && $this->assets[ $handle ] instanceof Script ) {
+					$this->add_extra_script_amp_dev_mode( $handle );
```

Comment:
Note that this should no longer be necessary after https://github.com/ampproject/amp-wp/pull/4001. As long as the dependency has the `ampdevmode` data added to it, there would be no more need to manually tag such inline scripts for matching a XPath expression.

Reply:
Thanks for reminding me of that! I've added two TODOs in the respective areas to remove our own fix in the future.